File: examples/rust/employee/src/employee.rs
use crate::utils::format_currency;
use std::fmt;

pub struct Employee {
    name: String,
    salary: f64,
}

impl Employee {
    // Constructor with arguments
    pub fn new(name: String, salary: f64) -> Employee {
        Employee { name, salary }
    }

    pub fn display_employee(&self) {
        println!("{}", self);
    }
}

impl fmt::Display for Employee {
    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
        let formatted_salary = format_currency(self.salary);
        write!(f, "Name: {}, Salary: {}", self.name, formatted_salary)
    }
}

File: examples/go/pkg/employee/employee.go
package employee

import (
	"fmt"

	"example.com/employee/pkg/utils"
)

type Employee struct {
	Name   string
	Salary float32
}

func (e Employee) String() string {
	formattedSalary := utils.FormatCurrency(float64(e.Salary))
	return fmt.Sprintf("Name: %s, Salary: %s", e.Name, formattedSalary)
}

func (e Employee) DisplayEmployee() {
	fmt.Println(e)
}

func NewEmployee(name string, salary float32) Employee {
	return Employee{
		Name:   name,
		Salary: salary,
	}
}
